compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/resourcepath/util/BallerinaNamedPathSegment.java

```diff
@@ -47,4 +46,15 @@ public Kind pathSegmentKind() {
     public String signature() {
         return this.name();
     }
+
+    @Override
+    @Deprecated
+    public String name() {
+        return this.getName().orElse("");
+    }
+
+    @Override
+    public Optional<String> getName() {
+        return Optional.of(super.getName().orElse(""));
```

Comment:
Can `super.getName()` be Optional of `null` here?

Reply:
`super.getName().orElse("")` added to remove the warning here

Reply:
No, I'm asking if it can be null? Why do we need to use empty string?